### ISSUE TITLE
Need to implement replication among different instances in a k8s installation

### ISSUE THREAD

**Issue by gbaru** [389ds/389-ds-base#5571]
I need to implement the replication among two or more Instances in a k8s infrastructure.

Instead to start alone and set up a partial solution, I would like to know if somebody is already doing something in this direction and possibly bring a contribute.

Thank you,
Gio

**Comment by Firstyear** [comment]
@vashirov Did you do any work in this space for automatic scaling?

**Comment by vashirov** [comment]
@gbaru, you can use StatefulSet for ordering and uniqueness guarantees of pods, and then set up replication as usual, see an example here: https://www.port389.org/docs/389ds/howto/howto-deploy-389ds-on-openshift.html

> @vashirov Did you do any work in this space for automatic scaling?

There was a plan to have it as part of [DS Operator](https://github.com/vashirov/ds-operator), but work on that was deprioritized. I know that Simon was interested in it, so we might get back to it eventually.

**Comment by Firstyear** [comment]
cc @droideck

**Comment by droideck** [comment]
Yeah, I was looking into that at some point, and surely it'll be fun to do!

So hopefully, soon I'll be able to look into that. But no promises or timeframe for now...

**Comment by gbaru** [comment]
> @gbaru, you can use StatefulSet for ordering and uniqueness guarantees of pods, and then set up replication as usual, see an example here: https://www.port389.org/docs/389ds/howto/howto-deploy-389ds-on-openshift.html
> 
> > @vashirov Did you do any work in this space for automatic scaling?
> 
> There was a plan to have it as part of [DS Operator](https://github.com/vashirov/ds-operator), but work on that was deprioritized. I know that Simon was interested in it, so we might get back to it eventually.

I am going to have a deep look in to that for sure.
My goals a limited at the moment. be able to start replication among containers in different K8s Nodes. 
But I am glad to contribute if a more broad attempt is underway.

**Comment by Firstyear** [comment]
@gbaru I'd say that the easiest way to start will probably be automating read-only replicas in k8s with an external data supplier. Getting auto-scaling on rw replicas will be more challenging.

**Comment by vashirov** [comment]
I'm going to close this for now. Work on replication as part of DS Operator will be tracked elsewhere.
